.travis.yml

```diff
@@ -36,6 +39,49 @@ matrix:
       env: PYTHON_VERSION=3.6 PIP_DEPENDENCIES='git+https://github.com/sphinx-doc/sphinx.git#egg=sphinx coveralls'
            CONDA_DEPENDENCIES="setuptools cython numpy pytest-cov"
 
+    # Test gcc on OSX
+    - os: osx
+      env:
+        - PYTHON_VERSION=3.5
+      before_install:
+        - rm /usr/local/include/c++
+        - brew install gcc6
+        - CC=gcc-6
+        - CXX=g++-6;
+
+    # Test manual install of LLVM's clang and LLVM-OpenMP
+    - os: osx
+      env:
+        - PYTHON_VERSION=3.5
+        - CONDA_DEPENDENCIES="setuptools sphinx cython numpy pytest-cov cmake"
+      before_install:
+        - wget http://releases.llvm.org/6.0.0/clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz
+        - tar -xvf clang+llvm-6.0.0-x86_64-apple-darwin.tar.xz
+        - loc=`pwd`
+        - path=$loc"/clang+llvm-6.0.0-x86_64-apple-darwin"
+        - export PATH=$path"/bin":$PATH
+        - export C_INCLUDE_PATH=$path"/include":$C_INCLUDE_PATH
+        - export LD_LIBRARY_PATH=$path"/lib":$LD_LIBRARY_PATH
+        - export CC=$path"/bin/clang"
+        - svn co http://llvm.org/svn/llvm-project/openmp/trunk openmp
+        - mkdir openmp-build && cd openmp-build
+        - cmake ../openmp
+        - make
+        - make install
+        - cd ../
```

Comment:
In order to keep the Travis config file as clean as possible, could you move all this into a script such as ``.install_llvm_clang.sh`` and just call that here?

Reply:
And actually I would recommend the same for the two other cases where you install things manually too - one script for each. This is just so that the install/setup cases are self contained and we can easily run them locally.